src/PackagePrivate/DatabaseTermIdsResolver.php

```diff
@@ -27,9 +27,6 @@ class DatabaseTermIdsResolver implements TermIdsResolver {
 	/** @var IDatabase */
 	private $dbw = null;
 
-	/** @var string[] stash of data returned from the {@link TypeIdsResolver} */
-	private $typeNames = [];
```

Comment:
This means that you might ask the `TypeIdsResolver` multiple times for the same type IDs if `resolveTermIds()` is called several times.

Reply:
yeap which seemed fine to me given that `resolveTermIds()` is not expected to be called more than once mostly or twice in some rare cases. right?

Reply:
> yeap which seemed fine to me given that `resolveTermIds()` is not expected to be called more than once mostly or twice in some rare cases. right?

Isn't this also going to be used in cases where we can't (easily) fetch batch fetch terms (like in the client when rendering pages that include Wikidata data)?

> TypeIdsResolver is expected to optimize asking for same types anyway in case it is used by many consumers over one request, so it should be also fine to leave this responsibility to it.

Sounds reasonable.

Reply:
I also expect there will be cases where `resolveTermIds()` is called several times. And sure, we can leave the caching to the `TypeIdsResolver`, but in my opinion it really doesn’t cost us much to keep the data around as well.

Reply:
hmm I'm half convinced now that this change isn't much better than how it was previously, so I'm fine discarding this purification attempt altogether.